pkg/controller/vitessbackupschedule/vitessbackupschedule_controller.go

```diff
@@ -0,0 +1,657 @@
+	}
+
+	meta.Annotations[scheduledTimeAnnotation] = scheduledTime.Format(time.RFC3339)
+
+	for k, v := range vbsc.Labels {
+		meta.Labels[k] = v
+	}
+
+	pod, err := r.createJobPod(ctx, vbsc, name)
+	if err != nil {
+		return nil, err
+	}
+	job := &kbatch.Job{
+		ObjectMeta: meta,
+		Spec: kbatch.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: meta,
+				Spec:       pod,
+			},
+		},
+	}
+
+	if err := ctrl.SetControllerReference(vbsc, job, r.scheme); err != nil {
+		return nil, err
+	}
+
+	return job, nil
+}
+
+func (r *ReconcileVitessBackupsSchedule) createJobPod(ctx context.Context, vbsc *planetscalev2.VitessBackupSchedule, name string) (pod corev1.PodSpec, err error) {
+	getVtctldServiceName := func(cluster string) (string, error) {
+		vtctldServiceName, vtctldServicePort, err := r.getVtctldServiceName(ctx, vbsc, cluster)
+		if err != nil {
+			return "", err
+		}
+		return fmt.Sprintf("--server=%s:%d", vtctldServiceName, vtctldServicePort), nil
+	}
+
+	// It is fine to not have any default in the event there is no strategy as the CRD validation
+	// ensures that there will be at least one item in this list. The YAML cannot be applied with
+	// empty list of strategies.
+	var cmd strings.Builder
+
+	addNewCmd := func(i int) {
+		if i > 0 {
+			cmd.WriteString(" && ")
+		}
+	}
+
+	for i, strategy := range vbsc.Spec.Strategy {
+		vtctldclientServerArg, err := getVtctldServiceName(vbsc.Spec.Cluster)
+		if err != nil {
+			return corev1.PodSpec{}, err
+		}
+
+		addNewCmd(i)
+		switch strategy.Name {
+		case planetscalev2.BackupShard:
+			if strategy.Keyspace == "" {
+				return pod, fmt.Errorf("the Keyspace field is missing from VitessBackupScheduleStrategy %s", planetscalev2.BackupShard)
+			}
+			if strategy.Shard == "" {
+				return pod, fmt.Errorf("the Shard field is missing from VitessBackupScheduleStrategy %s", planetscalev2.BackupShard)
+			}
+			createVtctldClientCommand(&cmd, vtctldclientServerArg, strategy.ExtraFlags, strategy.Keyspace, strategy.Shard)
+		case planetscalev2.BackupKeyspace:
+			if strategy.Keyspace == "" {
+				return pod, fmt.Errorf("the Keyspace field is missing from VitessBackupScheduleStrategy %s", planetscalev2.BackupKeyspace)
+			}
+			shards, err := r.getAllShardsInKeyspace(ctx, vbsc.Namespace, vbsc.Spec.Cluster, strategy.Keyspace)
+			if err != nil {
+				return corev1.PodSpec{}, err
+			}
+			for j, shard := range shards {
+				addNewCmd(j)
+				createVtctldClientCommand(&cmd, vtctldclientServerArg, strategy.ExtraFlags, strategy.Keyspace, shard)
+			}
+		case planetscalev2.BackupCluster:
+			keyspaces, err := r.getAllShardsInCluster(ctx, vbsc.Namespace, vbsc.Spec.Cluster)
+			if err != nil {
+				return corev1.PodSpec{}, err
+			}
+			for ksIndex, ks := range keyspaces {
+				for shardIndex, shard := range ks.shards {
+					if shardIndex > 0 || ksIndex > 0 {
+						cmd.WriteString(" && ")
+					}
+					createVtctldClientCommand(&cmd, vtctldclientServerArg, strategy.ExtraFlags, ks.name, shard)
```

Comment:
am i reading this right that it will be taking a backup of each keyspace and shard in sequence? that doesn't seem ideal to me because if each shard takes an hour to backup, and there are 32 shards, then the backup of the first shard and last shard will be more than a day apart.

i think it would be better if there were at least the _option_ of `BackupCluster` and `BackupKeyspace` to backup all keyspaces and shards in parallel.

might be better to limit this PR to only support `BackupShard` for now, and add support for the other options after more consideration into how to implement `BackupKeyspace` and `BackupCluster`.

Reply:
Let's do that, remove those two strategies as part of this PR and I will work on a subsequent PR to add them back with a better approach. This PR is getting lengthy already.

Reply:
Fixed via 70ba063

Reply:
IMO `BackupAllShardsInKeyspace` and `BackupAllShardsInCluster` are better names. It may seems nitty, but I think it's important as it reflects what it actually is: independent backups of the shards. i.e. it is NOT a single constant backup of the keyspace or cluster at any physical or logical point in time.

Reply:
I ended up removing Keyspace and Cluster strategies in this PR as it will require a bigger refactoring.